src/uwtools/templater.py

```diff
@@ -0,0 +1,109 @@
+    if not os.path.exists(arg):
+        msg = f'{arg} does not exist!'
+        raise argparse.ArgumentTypeError(msg)
+    return arg
+
+def parse_args(argv):
+
+    '''
+    Function maintains the arguments accepted by this script. Please see
+    Python's argparse documentation for more information about settings of each
+    argument.
+    '''
+
+    parser = argparse.ArgumentParser(
+       description='Update a Jinja2 Template with user-defined settings.'
+    )
+    parser.add_argument(
+        '-o', '--outfile',
+        help='Full path to output file',
+        )
+    parser.add_argument(
+        '-i', '--input_template',
+        help='Path to a Jinja2 template file.',
+        required=True,
+        type=file_exists,
+        )
+    parser.add_argument(
+        '-c', '--config_file',
+        help='Optional path to a YAML configuration file. If not provided, '
+        'os.environ is used to configure.',
+        type=file_exists,
+        )
+    parser.add_argument(
+        '-d', '--dry_run',
+        action='store_true',
+        help='If provided, print rendered template to stdout only',
+        )
+    parser.add_argument(
+        '--values_needed',
+        action='store_true',
+        help='If provided, print a list of required configuration settings to stdout',
+        )
+    return parser.parse_args(argv)
+
+def set_template(argv):
+    '''Main section for set_namelist ingest utility'''
+
+    cla = parse_args(argv)
+
+    if cla.config_file:
+        cfg = config.YAMLConfig(cla.config_file)
+    else:
+        cfg = os.environ
+
+
+    # instantiate Jinja2 environment and template
+    env = Environment(loader=FileSystemLoader(cla.input_template))
+    template = env.get_template('')
+
+    # Gather the undefined template variables
+    j2_parsed = env.parse(env.loader.get_source(env, ''))
+    undeclared_variables = meta.find_undeclared_variables(j2_parsed)
+
+    if cla.values_needed:
+        print('Values needed for this template are:')
+        for var in sorted(undeclared_variables):
+            print(var)
+
+        return
+
+    # Render the template with the specified config object
+    rendered_template = template.render(**cfg)
+
+    parser = f90nml.Parser()
+    # write out fully qualified and populated f90 namelist file
+    nml = parser.reads(rendered_template)
+
+    if cla.dry_run:
+        if cla.outfile:
+            print(f'warning file {cla.outfile} not written when using --dry_run or --values_needed')
+    # apply switch to allow user to view the results of namelist instead of writing to disk
+    if cla.dry_run:
+        print(nml)
+    # apply switch to print out required template values
+    # write out f90 name list
+    if not cla.dry_run and not cla.values_needed:
+        with open(cla.outfile, 'w+', encoding='utf-8') as nml_file:
+            f90nml.write(nml, nml_file)
+
+if __name__ == '__main__':
+    set_template(sys.argv[1:])
```

Comment:
not sure I understand this.
Is it a "generic" template substitution or specific to Fortran namelist?

Reply:
My understanding is the specific ticket this addresses is for the namelist, but it is anticipated that other tickets will be created to address other file types, and this will be extended to cover them in that work. @christinaholtNOAA do I have that correct?

Reply:
That is correct. I totally missed responding to this one! 

We do have work to do here to make this a generic templater where it is not currently. We definitely did not want to grow this work any larger than it already was.